grype/matcher/apk/matcher.go

```diff
@@ -135,3 +135,81 @@ func vulnerabilitiesByID(vulns []vulnerability.Vulnerability) map[string][]vulne
 
 	return results
 }
+
+func (m *Matcher) findApkPackage(store vulnerability.Provider, d *distro.Distro, p pkg.Package) ([]match.Match, error) {
+	var matches = make([]match.Match, 0)
+
+	// find Alpine SecDB matches for the given package name and version
+	secDbMatches, err := common.FindMatchesByPackageDistro(store, d, p, m.Type())
+	if err != nil {
+		return nil, err
+	}
+
+	cpeMatches, err := m.cpeMatchesWithoutSecDbFixes(store, d, p)
+	if err != nil {
+		return nil, err
+	}
+
+	// keep all secdb matches, as this is an authoritative source
+	matches = append(matches, secDbMatches...)
+
+	// keep only unique CPE matches
+	matches = append(matches, deduplicateMatches(secDbMatches, cpeMatches)...)
+
+	return matches, nil
+}
+
+func (m *Matcher) matchBySourceIndirection(store vulnerability.Provider, d *distro.Distro, p pkg.Package) ([]match.Match, error) {
+	metadata, ok := p.Metadata.(pkg.ApkMetadata)
+	if !ok {
+		return nil, nil
+	}
+
+	// ignore packages without source indirection hints or where source name is identical to package name
+	if metadata.OriginPackage == "" || metadata.OriginPackage == p.Name {
+		return []match.Match{}, nil
```

Comment:
```suggestion
		return nil, nil
```

(empty slices in Go are expressed as `nil`, unless there's a special reason for an allocated instance)